pkg/kvstore/etcd.go

```diff
@@ -822,59 +803,13 @@ func makeSessionName(sessionPrefix string, opts *ExtraOptions) string {
 	return sessionPrefix
 }
 
-func getEPVersion(ctx context.Context, c client.Maintenance, etcdEP string, timeout time.Duration) (semver.Version, error) {
-	ctxTimeout, cancel := context.WithTimeout(ctx, timeout)
-	defer cancel()
-	sr, err := c.Status(ctxTimeout, etcdEP)
-	if err != nil {
-		return semver.Version{}, Hint(err)
-	}
-	v, err := versioncheck.Version(sr.Version)
-	if err != nil {
-		return semver.Version{}, fmt.Errorf("error parsing server version %q: %s", sr.Version, Hint(err))
-	}
-	return v, nil
-}
-
 func (e *etcdClient) sessionError() (err error) {
 	e.RWMutex.RLock()
 	err = e.sessionErr
 	e.RWMutex.RUnlock()
 	return
 }
 
-// checkMinVersion checks the minimal version running on etcd cluster.  This
-// function should be run whenever the etcd client is connected for the first
-// time and whenever the session is renewed.
-func (e *etcdClient) checkMinVersion(ctx context.Context, timeout time.Duration) error {
```

Comment:
:thinking: Are you thinking about checking that at installation time or when doing `cilium status`? The main issue I see is that it is tricky to ensure that the CLI is always able to access the etcd members, independently of whether they are outside or inside the cluster. And I'm not 100% sure whether the effort is really worth considering that Cilium is currently compatible with any supported version of etcd.

Reply:
I was thinking about doing that in `cilium status`. For now, just creating issue in cilium-cli to capture that we might want to introduce this check. Even though we support all versions, I think it would be nice to suggest upgrades etc in case users are running Etcd older than 3.4, or 3.5 with [known data inconsistency issues](https://groups.google.com/a/kubernetes.io/g/dev/c/B7gJs88XtQc/m/rSgNOzV2BwAJ) (versions 3.5.0-3.5.2)

Reply:
https://github.com/cilium/cilium-cli/issues/2107